#!/usr/bin/env python
PACKAGE = "sensor_fusion"

# Import the parameter generator for dynamic_reconfigure
from dynamic_reconfigure.parameter_generator_catkin import *

# Create an instance of the ParameterGenerator
gen = ParameterGenerator()

# Add parameters to the configuration
# Each parameter has a name, type, level, description, default value, and range (min, max)

# Add a double parameter named "z_hit"
gen.add("z_hit", double_t, 0, "Weight for hit probability", 0.5, 0, 5)

# Add a double parameter named "z_short"
gen.add("z_short", double_t, 0, "Weight for short probability", 0.5, 0, 5)

# Add a double parameter named "z_max"
gen.add("z_max", double_t, 0, "Weight for max probability", 0.5, 0, 5)

# Add a double parameter named "z_rand"
gen.add("z_rand", double_t, 0, "Weight for random probability", 0.5, 0, 5)

# Add an integer parameter named "visualize_rays_percentage"
gen.add("visualize_rays_percentage", int_t, 0, "Percentage of rays to visualize", 10, 0, 100)

# Add a double parameter named "sigma_hit"
gen.add("sigma_hit", double_t, 0, "Standard deviation for hit probability", 0.1, 0.0, 1.0)

# Add a double parameter named "lambda_short"
gen.add("lambda_short", double_t, 0, "Lambda for short probability", 0.1, 0.0, 1.0)

# Generate the configuration file for dynamic_reconfigure
exit(gen.generate(PACKAGE, "sensor_fusion", "SensorModel"))     # Name of the node, which is called by #include <sensor_fusion/SensorModelConfig.h>